pkg/model/gcemodel/context.go

```diff
@@ -54,7 +54,11 @@ func (c *GCEModelContext) NameForIPAliasRange(key string) string {
 func (c *GCEModelContext) LinkToSubnet(subnet *kops.ClusterSubnetSpec) *gcetasks.Subnet {
 	name := subnet.ProviderID
 	if name == "" {
-		name = c.SafeObjectName(subnet.Name)
+		var err error
+		name, err = gce.ClusterSuffixedName(subnet.Name, c.Cluster.ObjectMeta.Name, 63)
```

Comment:
(Note to self: I validated that ClusterSuffixedName does not change names unless we would be exceeding the length, i.e. this is not breaking)